src/parsec.rs

```diff
@@ -224,3 +284,160 @@ fn new_rng() -> Box<dyn Rng> {
+    }
+
+    fn create_full_ids() -> Vec<FullId> {
+        (0..DEFAULT_MIN_SECTION_SIZE)
+            .map(|_| FullId::new())
+            .collect()
+    }
+
+    fn create_gen_pfx_info(full_ids: Vec<FullId>, version: u64) -> GenesisPfxInfo {
+        let members = full_ids.iter().map(|id| *id.public_id()).collect();
+        let section_info = unwrap!(SectionInfo::new_for_test(
+            members,
+            Prefix::<XorName>::default(),
+            version
+        ));
+        GenesisPfxInfo {
+            first_info: section_info,
+            first_state_serialized: Vec::new(),
+            latest_info: SectionInfo::default(),
+        }
+    }
+
+    // Create a ParsecMap with 3 sections
+    fn create_parsec_map(size: u64) -> ParsecMap {
+        let log_ident = LogIdent::new("node");
+        let full_ids = create_full_ids();
+        let full_id = full_ids[0].clone();
+
+        let gen_pfx_info = create_gen_pfx_info(full_ids.clone(), 0);
+        let mut parsec_map = ParsecMap::new(full_id.clone(), &gen_pfx_info);
+
+        for parsec_no in 1..=size {
+            let gen_pfx_info = create_gen_pfx_info(full_ids.clone(), parsec_no);
+            parsec_map.init(full_id.clone(), &gen_pfx_info, &log_ident);
+        }
+
+        parsec_map
+    }
+
+    trait HandleRequestResponse {
+        fn handle(
+            &self,
+            parsec_map: &mut ParsecMap,
+            msg_version: u64,
+            pub_id: &id::PublicId,
+            log_ident: &LogIdent,
+        );
+    }
+
+    impl HandleRequestResponse for Request {
+        fn handle(
+            &self,
+            parsec_map: &mut ParsecMap,
+            msg_version: u64,
+            pub_id: &id::PublicId,
+            log_ident: &LogIdent,
+        ) {
+            let _ = parsec_map.handle_request(msg_version, self.clone(), *pub_id, &log_ident);
+        }
+    }
+
+    impl HandleRequestResponse for Response {
+        fn handle(
+            &self,
+            parsec_map: &mut ParsecMap,
+            msg_version: u64,
+            pub_id: &id::PublicId,
+            log_ident: &LogIdent,
+        ) {
+            let _ = parsec_map.handle_response(msg_version, self.clone(), *pub_id, &log_ident);
+        }
+    }
+
+    fn handle_msgs_just_below_prune_limit<T: HandleRequestResponse + Serialize>(
+        parsec_map: &mut ParsecMap,
+        msg_version: u64,
+        msg: &T,
+        pub_id: &id::PublicId,
+        log_ident: &LogIdent,
+    ) {
+        let msg_size = serialisation::serialised_size(&msg);
+        let msg_size_limit = PARSEC_SIZE_LIMIT / msg_size;
+
+        // Handle msg_size_limit msgs which should trigger pruning needed on the next
+        // msg if it's against the latest parsec.
+        for _ in 0..msg_size_limit {
+            msg.handle(parsec_map, msg_version, pub_id, log_ident);
+        }
+
+        // Make sure we don't cross the prune limit
+        assert_eq!(parsec_map.needs_pruning(), false);
+    }
+
+    #[allow(unused)]
```

Comment:
Why allow unsused?

Reply:
Oops that one snuck through the cracks, thanks for spotting!

Reply:
Removed